frontend/src/components/dashboard/DashboardHeader.jsx

```diff
@@ -1,73 +1,174 @@
 import Link from "next/link";
-
-import { Row, Col } from "react-bootstrap";
+import {
+  Button,
+  Col,
+  Dropdown,
+  FormControl,
+  InputGroup,
+  Row
+} from "react-bootstrap";
 
 import CardCell from "../utils/CardCell";
+import AccountApi from "../../libraries/explorer-wamp/accounts";
+import BlocksApi from "../../libraries/explorer-wamp/blocks";
+import TransactionsApi from "../../libraries/explorer-wamp/transactions";
+
+import Router from "next/router";
+
+export class DashboardHeader extends React.Component {
+  state = {
+    searchValue: ""
+  };
+
+  handleSearch = async event => {
+    event.preventDefault();
+
+    const { searchValue } = this.state;
+
+    const [block, transaction, account] = await Promise.all([
+      new BlocksApi().getBlockInfo(searchValue).catch(() => {}),
+      new TransactionsApi().getTransactionInfo(searchValue).catch(() => {}),
+      new AccountApi().queryAccount(searchValue).catch(() => {})
+    ]);
+
+    if (block) {
+      return Router.push("/blocks/" + searchValue);
+    }
+    if (transaction && transaction.signerId) {
+      return Router.push("/transactions/" + searchValue);
+    }
+    if (account) {
+      return Router.push("/accounts/" + searchValue);
+    }
 
-export default ({
-  onlineNodesCount,
-  totalNodesCount,
-  lastBlockHeight,
-  transactionsPerSecond,
-  lastDayTxCount,
-  accountsCount
-}) => (
-  <div className="dashboard-info-container">
-    <Row noGutters>
-      <Col xs="12" md="3">
-        <Link href="/nodes">
-          <a>
+    alert("Nothing found!");
```

Comment:
This is not pretty, but I prefer working on tasks incrementally instead of blocking them forever in PRs, so I would create a separate issue to address the error reporting.